core/include/scipp/core/dataset.h

```diff
@@ -547,7 +548,10 @@ class SCIPP_CORE_EXPORT DatasetConstProxy {
         indices.end());
     // The dimension extent is either given by the coordinate, or by data, which
     // can be 1 shorter in case of a bin-edge coordinate.
-    scipp::index extent = coords()[slice1.dim].dims()[slice1.dim];
+    scipp::index extent =
+        coords()[slice1.dim].dims()[slice1.dim]; // TODO this is a bug. Cannot
+                                                 // rely on getting extents via
+                                                 // coords
```

Comment:
Can you elaborate?

Reply:
If you create a dataset with only data variables added (no coordinates).